app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php

```diff
@@ -167,12 +167,12 @@ public function execute()
             $attributeCode = $attributeCode ?: $this->generateCode($this->getRequest()->getParam('frontend_label')[0]);
             if (strlen($attributeCode) > 0) {
                 $validatorAttrCode = new \Zend_Validate_Regex(
-                    ['pattern' => '/^[a-z\x{600}-\x{6FF}][a-z\x{600}-\x{6FF}_0-9]{0,30}$/u']
+                    ['pattern' => '/^[a-zA-Z\x{600}-\x{6FF}][a-zA-Z\x{600}-\x{6FF}_0-9]{0,30}$/u']
```

Comment:
This is not the only place this kind of validation happens.
I know that this PR is for creating a product in admin but I think that we should be consistent with validation and apply it also to Web API, that is, in `\Magento\Catalog\Model\Product\Attribute\Repository::validateCode()`